### PR TITLE
[max] Fixed an error sending a QuantityType command

### LINKED ISSUE #4119: [max!] Autorefresh don't work anymore
## Expected Behavior
Autorefresh of temperature works in the configured intervall.

## Current Behavior
No refresh in this Intervall.

## Steps to Reproduce (for Bugs)
1. Configure Thermostat things with a refreshrate >10
2. set Loglevel to TRACE for org.openhab.binding.max binding

## Context
I have update from 2.3 to 2.4.0.m4.
All actuell temeratures from my devices are logged and shown in a chart, but after upgarding this don't work anymore. I reset the cube and reconnect all devices. Also i tried to delete all max things from openhab and reinstall.
I've tried different refresh intervalls from 10,11,12,13 Minutes to 60,70,80,90 which worked before.
During summertime with all devices set to manual (with OH2.3) I recognised that my room 1+2 are update correct but in room 3+4 somtimes the temperature isn't set back correctly, So it grows day by day and must be reset manualy to eco temperature.

## Your Environment
* Openhab 2.4.0.m4
* Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
* Raspian strech
* Max cube v1.13
* 2x MAX! Elektronischer Heizkörperthermostat (room 1+2)
* 2x MAX!-Fensterkontakt (room 1+2)
* 2x MAX! Heizkörperthermostat basic (room 3+4)

### PR DESCRIPTION
**Pull request by cweitkamp**
- Fixed an error sending a QuantityType command
- Minor code clean-ups

Closes #4119

Signed-off-by: Christoph Weitkamp <github@christophweitkamp.de>

### CODE REVIEW

File: addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java [hunk 1]
```diff
@@ -405,51 +407,43 @@ private CubeCommand getCommand(SendCommand sendCommand) {
...
-
         // Temperature setting
         if (channelUID.getId().equals(CHANNEL_SETTEMP)) {
-            if (command instanceof DecimalType || command instanceof OnOffType) {
-                DecimalType decimalType = DEFAULT_OFF_TEMPERATURE;
+            if (command instanceof DecimalType || command instanceof QuantityType || command instanceof OnOffType) {
+                double setTemp = DEFAULT_OFF_TEMPERATURE;
                 if (command instanceof DecimalType) {
-                    decimalType = (DecimalType) command;
+                    setTemp = ((DecimalType) command).doubleValue();
```
Comment (wborn) [inline addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: DecimalType commands will be automatically converted to QuantityType commands (https://github.com/eclipse/smarthome/pull/5734). 
So all logic for handling DecimalType commands on QuantityType channels in bindings will be dead code.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: Yes, I know that. But what if the user creates a basic `Number` item? Should we be backward compatible or not?

Comment (wborn) [reply addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: It seems to be backwards compatible too when I create a `Number` item without a dimension for the `set_point` channel of a Nest Thermostat (Nest Binding). When I change this item via a sitemap, the handler receives a `QuantityType` command.

Comment (cweitkamp) [reply addons/binding/org.openhab.binding.max/src/main/java/org/openhab/binding/max/internal/handler/MaxCubeBridgeHandler.java]: Alright. I trust you and I have changed it.

**Pull request merged**